File: db4o.net/drs/Db4objects.Drs.Tests/Db4objects.Drs.Tests/Foundation/SignatureTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4oUnit;
using Db4objects.Db4o.Foundation;
using Db4objects.Db4o.Internal;
using Db4objects.Db4o.Internal.Encoding;
using Db4objects.Drs.Foundation;

namespace Db4objects.Drs.Tests.Foundation
{
	public class SignatureTestCase : ITestCase
	{
		public virtual void Test()
		{
			StatefulBuffer writer = new StatefulBuffer(null, 300);
			string stringRepresentation = SignatureGenerator.GenerateSignature();
			new LatinStringIO().Write(writer, stringRepresentation);
			Signature signature = new Signature(writer.GetWrittenBytes());
			Assert.AreEqual(stringRepresentation, signature.ToString());
		}
	}
}

File: drs/src/test/com/db4o/drs/test/foundation/SignatureTestCase.java
/* Copyright (C) 2004 - 2010  Versant Inc.  http://www.db4o.com */

package com.db4o.drs.test.foundation;

import com.db4o.drs.foundation.*;
import com.db4o.foundation.*;
import com.db4o.internal.*;
import com.db4o.internal.encoding.*;

import db4ounit.*;

public class SignatureTestCase implements TestCase {
	
	public void test(){
        StatefulBuffer writer = new StatefulBuffer(null, 300);
        String stringRepresentation = SignatureGenerator.generateSignature();
		new LatinStringIO().write(writer, stringRepresentation);
		Signature signature = new Signature(writer.getWrittenBytes());
		Assert.areEqual(stringRepresentation, signature.toString());
	}

}
